src/EnergyPlus/InputProcessing/CsvParser.cc

```diff
@@ -212,33 +222,71 @@ void CsvParser::parse_line(std::string_view csv, size_t &index, json &columns)
+        size_t found_index = csv.find_first_of("\r\n", this_beginning_of_line_index);
+        std::string_view line;
+        if (found_index != std::string::npos) {
+            line = csv.substr(this_beginning_of_line_index, found_index - this_beginning_of_line_index);
+        }
+        return line;
+    };
+
     while (true) {
         token = look_ahead(csv, index);
         if (token == Token::LINE_END || token == Token::FILE_END) {
-            if (parsed_values != num_columns) {
+            if (has_extra_columns) {
+                warnings_.emplace_back(
+                    fmt::format("CsvParser - Line {} - Expected {} columns, got {}. Ignored extra columns. Error in following line.",
+                                this_cur_line_num,
+                                num_columns,
+                                parsed_values),
+                    false);
+                warnings_.emplace_back(getCurrentLine(), true);
+            } else if (parsed_values != num_columns) {
                 success = false;
 
-                size_t found_index = csv.find_first_of("\r\n", this_beginning_of_line_index);
-                std::string line;
-                if (found_index != std::string::npos) {
-                    line = csv.substr(this_beginning_of_line_index, found_index - this_beginning_of_line_index);
-                }
                 errors_.emplace_back(
                     fmt::format(
                         "CsvParser - Line {} - Expected {} columns, got {}. Error in following line.", this_cur_line_num, num_columns, parsed_values),
                     false);
-                errors_.emplace_back(line, true);
+                errors_.emplace_back(getCurrentLine(), true);
             }
             next_token(csv, index);
             return;
         } else if (token == Token::DELIMITER) {
             next_token(csv, index);
+            token = look_ahead(csv, index);
+            if (token == Token::DELIMITER) {
+                // Two delimiters in a row means a blank value
+                // This is not yet an error, in case the user is not using this column... It will crash later if they do try to cast it to a number
+                size_t const next_col = column_num + 1;
+                if (next_col < num_columns) {
+                    // Push a nan for blank value
+                    columns.at(next_col).push_back(json::value_t::null);
+                    warnings_.emplace_back(fmt::format("CsvParser - Line {} Column {} - Blank value found, setting to null. Error in following line.",
+                                                       this_cur_line_num,
+                                                       next_col + 1),
+                                           false);
+                    warnings_.emplace_back(getCurrentLine(), true);
+                } else {
+                    has_extra_columns = true;
+                }
+                ++parsed_values;
+            }
```

Comment:
In the Delimiter case, we scan ahead to check if another delimiter is coming up, in which case we emplace a null.

This is a warning, because unless you actually try to use that column, it's fine.